streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -174,6 +200,32 @@ public void process(final Record<KIn, VIn> record) {
                     keyClass,
                     valueClass),
                 e);
+        } catch (final FailedProcessingException | TaskCorruptedException | TaskMigratedException e) {
+            // Rethrow exceptions that should not be handled here
+            throw e;
+        } catch (final Exception e) {
+            final ErrorHandlerContext errorHandlerContext = new DefaultErrorHandlerContext(
+                internalProcessorContext.topic(),
+                internalProcessorContext.partition(),
+                internalProcessorContext.offset(),
+                internalProcessorContext.headers(),
+                internalProcessorContext.recordContext().rawRecord().key(),
+                internalProcessorContext.recordContext().rawRecord().value(),
+                internalProcessorContext.currentNode().name(),
+                internalProcessorContext.taskId());
+
+            final ProcessingExceptionHandler.ProcessingHandlerResponse response = processingExceptionHandler
+                .handle(errorHandlerContext, record, e);
+
+            if (response == ProcessingExceptionHandler.ProcessingHandlerResponse.FAIL) {
+                log.error("Processing exception handler is set to fail upon" +
+                     " a processing error. If you would rather have the streaming pipeline" +
+                     " continue after a processing error, please set the " +
+                     PROCESSING_EXCEPTION_HANDLER_CLASS_CONFIG + " appropriately.");
+                throw new FailedProcessingException(e);
+            } else {
+                droppedRecordsSensor.record();
```

Comment:
Should we log here? How do we do this elsewhere? I know that the default handlers log already (so might be redundant), but a custom handler might not log...? Just curious.

Reply:
I also thought about logging here. But since we had quite some issues with log pressure due to dropped records, I thought, it would be better to let the handler log.